scripts/lib.sh

```diff
@@ -119,6 +119,13 @@ install-py-dev() {
         # https://github.com/unbit/uwsgi/issues/2361
         pip install https://storage.googleapis.com/python-arm64-wheels/uWSGI-2.0.19.1-cp38-cp38-macosx_11_0_universal2.whl
     fi
+    # This hack is until we can upgrade to a newer version of Selenium
+    fx_profile=.venv/lib/python3.8/site-packages/selenium/webdriver/firefox/firefox_profile.py
+    # Remove this block when upgrading the selenium package
+    if grep -q "or setting is" "${fx_profile}"; then
+        echo "We are patching ${fx_profile}. You will see this message only once."
+        patch -p0 <scripts/patches/firefox_profile.diff
```

Comment:
Instead of guarding with the grep, you can idempotently (`--forward`) apply the patch like this: `patch -p0 --forward -r=/dev/null < scripts/patches/firefox_profile.diff`. The `-r=/dev/null` is just to prevent it writing a reject file in the case of failure, but it still exits nonzero.

Reply:
Oh hmm, it still exits 1. Never mind.

Reply:
https://unix.stackexchange.com/questions/603874/can-patch-be-idempotent